### PR TITLE
Add drumbeat ticker to FMv2

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/services/fluxmonitorv2/flux_monitor.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const (
 	PollRequestTypeHibernation
 	PollRequestTypeRetry
 	PollRequestTypeAwaken
+	PollRequestTypeDrumbeat
```
Comment (jkongie) [inline core/services/fluxmonitorv2/flux_monitor.go:48]: What about `pulse`?

File: core/internal/cltest/cltest.go [hunk 2]
```diff
@@ -1718,11 +1718,17 @@ func NewAwaiter() Awaiter { return make(Awaiter) }
 
 func (a Awaiter) ItHappened() { close(a) }
 
-func (a Awaiter) AssertHappened(t *testing.T) {
+func (a Awaiter) AssertHappened(t *testing.T, expected bool) {
```
Comment (samsondav) [inline core/internal/cltest/cltest.go:1721]: `AssertDidntHappen` ?

**Review by jkongie — COMMENTED** [review]
Just have a suggestion on the `heartbeat` name but the implementation is good.

File: core/store/migrations/0046_add_fmv2_drumbeat_ticker.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package migrations
+
+import (
+	"gorm.io/gorm"
+)
+
+const up46 = `
+    ALTER TABLE flux_monitor_specs ADD COLUMN drumbeat_enabled boolean DEFAULT false;
```
Comment (samsondav) [inline core/store/migrations/0046_add_fmv2_drumbeat_ticker.go:8]: Let's set this `NOT NULL` which we are here

**Review by samsondav — APPROVED** [review]

**Comment by samsondav** [comment]
Does it rely on the nodes having synchronized clocks?

**Review by samsondav — APPROVED** [review]

**Pull request merged**